### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by Thryphore**

### CODE REVIEW

**Comment by Thryphore** [comment]
Can't be updated any further, as react-bootstrap want react 17.0.2 whereas everything else that can't be updated any further wants the newest version of react. A couple different things want typescript 4, so it's updated to the newest version of 4.

File: frontend/src/pages/devices/Detail.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React from "react"
 import { Alert, Button, Form, InputGroup } from "react-bootstrap"
 import { format, formatDistance, parseISO } from "date-fns"
-import { useHistory, useRouteMatch } from "react-router-dom"
+import { useNavigate, useMatch } from "react-router-dom"
 import { LockCode } from "./components/LockCode"
 import { Loading } from "../utils/Loading"
 import { StandardFetch } from "../utils/FetchHelper"
-
+console.log("we get to devices")
```
Comment (anthonywittig) [inline frontend/src/pages/devices/Detail.tsx:8]: You can kill this ;)

File: frontend/src/pages/devices/Detail.tsx [hunk 2]
```diff
@@ -50,10 +50,10 @@ const Detail = () => {
     setRevision(revision + 1)
   }
 
-  const m = useRouteMatch("/" + Endpoint + "/:id")
```
Comment (anthonywittig) [inline frontend/src/pages/devices/Detail.tsx:53]: Using the `Endpoint` variable can help with making sure we're consistent if we ever change our route (or more likely, copy/paste this code for a new detail page).

File: frontend/src/pages/devices/Detail.tsx [hunk 3]
```diff
@@ -246,7 +246,7 @@ const Detail = () => {
               disabled={true}
               aria-describedby="basic-addon2"
             />
-            <InputGroup.Append>
```
Comment (anthonywittig) [inline frontend/src/pages/devices/Detail.tsx:249]: Can we add the append back? It looks like the button is on it's own row now.

File: frontend/src/pages/units/Detail.tsx [hunk 4]
```diff
@@ -38,9 +40,9 @@ export const Detail = () => {
   const [properties, setProperties] = React.useState<Property[]>([])
   const [devices, setDevices] = React.useState<DeviceT[]>([])
   const [reservations, setReservations] = React.useState<Reservation[]>([])
-  const history = useHistory()
+  const navigate = useNavigate()
 
-  const m = useRouteMatch("/" + Endpoint + "/:id")
```
Comment (anthonywittig) [inline frontend/src/pages/units/Detail.tsx:43]: Could use `Endpoint` here.

**Review by anthonywittig — COMMENTED** [review]
Nice work!

**Review by anthonywittig — DISMISSED** [review]
I think this is good to merge now. I'll add a follow up issue for any styling updates I think we should look at.

**Comment by anthonywittig** [comment]
Actually, let's make one more change...

We're currently pulling in bootstrap@4.5.3 (search the codebase to see where that's happening). We should pull in whatever version `react-bootstrap` wants us to use.

File: wireframes/dashboard.html [hunk 5]
```diff
@@ -6,7 +6,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
 
     <!-- Bootstrap CSS -->
-    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">
```
Comment (anthonywittig) [inline wireframes/dashboard.html:9]: The `wireframes` directory was just to demo what we were going to build; if you felt motivated you could delete it now.

**Review by anthonywittig — COMMENTED** [review]
image


All the rows of our forms need some margin. It looks like all our `<Form.Group>` elements needs a `className="mb-3"` ([docs](https://react-bootstrap.netlify.app/docs/forms/overview)).

**Review by anthonywittig — APPROVED** [review]

**Pull request merged**